server/src/main/java/io/druid/server/QueryResource.java

```diff
@@ -167,19 +167,19 @@ public Response getServer(@PathParam("id") String queryId, @Context final HttpSe
   public Response doPost(
       InputStream in,
       @QueryParam("pretty") String pretty,
-      @Context final HttpServletRequest req // used to get request content-type, remote address and AuthorizationInfo
+      @Context final HttpServletRequest req // used to get request content-type,Accept header, remote address and AuthorizationInfo
   ) throws IOException
   {
     final long start = System.currentTimeMillis();
     Query query = null;
     QueryToolChest toolChest = null;
     String queryId = null;
 
-    final ResponseContext context = createContext(req.getContentType(), pretty != null);
+    final ResponseContext context = createContext(req.getHeader("Accept"), pretty != null);
```

Comment:
To retain prior behavior this should default to the content type if the `Accept` header is missing